app/src/main/java/com/javahelp/frontend/activity/ProviderRegistrationViewmodel.java

```diff
@@ -1,17 +1,199 @@
 package com.javahelp.frontend.activity;
 
 import android.app.Application;
-import android.view.View;
 
 import androidx.annotation.NonNull;
 import androidx.lifecycle.AndroidViewModel;
 import androidx.lifecycle.MutableLiveData;
-import androidx.lifecycle.ViewModel;
 
-public class ProviderRegistrationViewmodel extends AndroidViewModel {
+
+import com.javahelp.frontend.domain.user.register.IRegisterDataAccess;
+import com.javahelp.frontend.domain.user.register.IRegisterOutput;
+import com.javahelp.frontend.domain.user.register.ISaltDataAccess;
+import com.javahelp.frontend.domain.user.register.RegisterInteractor;
+import com.javahelp.frontend.domain.user.register.RegisterResult;
+
+import com.javahelp.frontend.gateway.LambdaRegisterDataAccess;
+import com.javahelp.frontend.gateway.LambdaSaltDataAccessProvider;
+
+import com.javahelp.model.token.Token;
+import com.javahelp.model.user.IPasswordHasher;
+import com.javahelp.model.user.ProviderUserInfo;
+import com.javahelp.model.user.SHAPasswordHasher;
+import com.javahelp.model.user.User;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class ProviderRegistrationViewmodel extends AndroidViewModel implements IRegisterOutput {
+
+    private MutableLiveData<String> username = new MutableLiveData<>("");
+    private MutableLiveData<String> password1 = new MutableLiveData<>("");
+    private MutableLiveData<String> password2 = new MutableLiveData<>("");
+    private MutableLiveData<String> email = new MutableLiveData<>("");
+    private MutableLiveData<String> firstname = new MutableLiveData<>("");
```

Comment:
Yeah, these two `firstname` and `lastname` should become `practicename` instead.